util/jwt/cjwt/cjwt.go

```diff
@@ -0,0 +1,98 @@
+// The package provides a way to create compact JWTs, "cJWT".
+//
+// cJWTs for JWT with a large payload can be 1/3 the size of the original.
+// This is achieved by compressing the payload before base 64 encoding it.
+//
+// cJWTs are only compact if they are smaller than the original JWTs.
+// as long as they are smaller than the cJWT representation.
+//
+// To help differentiate,
+// * JWT have 3 dots. cJWTs have 4 dots.
+// * cJWTs start with "cJWT/v1:"
+//
+package cjwt
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"strings"
+)
+
+var encoding = base64.RawStdEncoding
+var magicNumber = "cJWT/v1:"
+
+// CompactJWT JWT to either a cJWT or return the original JWT, whichever is smaller.
+func CompactJWT(text string) (string, error) {
+	parts := strings.SplitN(text, ".", 3)
+	if len(parts) != 3 {
```

Comment:
I wonder if we should also only compress them if the existing JWT might be too big for a cookie as well. e.g. `len(text)` > 4093-someBuffer`